tests/ci_tests/test_coffee_maker_mains.py

```diff
@@ -0,0 +1,45 @@
+"""Tests ensuring each coffee_maker script with a __main__ guard imports cleanly."""
+
+from __future__ import annotations
+
+import os
+import runpy
+from pathlib import Path
+
+import pytest
+
+_REPO_ROOT = Path(__file__).resolve().parents[1].parent
+_COFFEE_MAKER_DIR = _REPO_ROOT / "coffee_maker"
+
+
+def _discover_main_scripts() -> list[Path]:
+    scripts: list[Path] = []
+    for candidate in _COFFEE_MAKER_DIR.rglob("*.py"):
+        try:
+            source = candidate.read_text(encoding="utf-8")
+        except OSError:
+            continue
+        if 'if __name__ == "__main__"' in source:
+            scripts.append(candidate)
+    return scripts
```

Comment:
Added docstring for the function 'is_palindrome'.
```suggestion
def is_palindrome(input_string: str) -> bool:
```